src/twisted/words/protocols/jabber/component.py

```diff
@@ -23,7 +23,7 @@
 from twisted.application import service
 from twisted.internet import defer
 from twisted.python import log
-from twisted.python.compat import _coercedUnicode, unicode
+from twisted.python.compat import unicode
```

Comment:
`unicode` can go, and be replaced by `str`.  You can do that in this PR, or a separate one if you want.